### ISSUE TITLE
Fix false positives for interpolation in keyframes-name-pattern

### ISSUE THREAD

**Issue by jdufresne** [stylelint/stylelint#5997]
### What steps are needed to reproduce the bug?

test.scss

```scss
$frames: ("one", "two", "three");

@each $name in $frames {
  @keyframes frame-#{$name} {
    /* CSS ... */
  }
}
```

```
$ npx stylelint test.scss

test.scss
 4:14  ✖  Expected keyframe name to be kebab-case  keyframes-name-pattern
```

The name is kebab case, but due to the string interpolation, the line is flagged as an error.

This was originally file at https://github.com/stylelint-scss/stylelint-config-standard-scss/issues/16 but I was instructed to move it here.

### What Stylelint configuration is needed to reproduce the bug?

```json
{
  "extends": "stylelint-config-standard-scss"
}
```

### How did you run Stylelint?

npx stylelint test.scss

### Which version of Stylelint are you using?

`14.6.1`

### What did you expect to happen?

No problems to be reported.

### What actually happened?

```
test.scss
 4:14  ✖  Expected keyframe name to be kebab-case  keyframes-name-pattern
```

### Does the bug relate to non-standard syntax?

Yes it uses SCSS string interpolation.

**Comment by jeddy3** [comment]
@jdufresne Thanks for moving the issue over and using the template.


I suggest we:
- broaden the scope of the [`isStandardSyntaxMediaFeatureName`](https://github.com/stylelint/stylelint/blob/main/lib/utils/isStandardSyntaxMediaFeatureName.js) utility to all names (and rename it `isStandardSyntaxName`)
- update the utility to replace the code for interpolation with the [`hasInterpolation`](https://github.com/stylelint/stylelint/blob/main/lib/utils/hasInterpolation.js) utility
- use the `isStandardSyntaxName` utility in the `keyframes-name-pattern` rule

I've labelled the issue as ready to implement. Please consider [contributing](https://stylelint.io/contributing) if you have time.

There are [steps on how to fix a bug in a rule](https://stylelint.io/developer-guide/rules#fix-a-bug-in-a-rule) in the Developer guide.

**Comment by fakoredeDamilola** [comment]
hello I will like to try this

**Comment by jeddy3** [comment]
@fakoredeDamilola That'd be great. I'll mark the issue as work in progress.

**Comment by a-falkonia** [comment]
@jeddy3 Hi, I'd love to work on the issue if this is still relevant.

**Comment by jeddy3** [comment]
@a-falkonia Yes [it's still relevant](https://stylelint.io/demo#N4Igxg9gJgpiBcIAkAzATgQwLYwM7wAIAKAHRAgDsYyAaAsgFwHcJb6QGALNGakASgDcJCiIACMDGE4EkFbDAIBLCrPQLcBYCIIExAaxgBPdTk2mYAWgDEwOQoC+WnboIB6AFQEAwgGVfBAB0wQQebi4OIg4gNCC4RhQMGAAeCHFguLgx4JQoSgDmadqq7GgArgA2eGSExa7shiaYZpbyOJYADhgMDDBooggEANou9WQAekRDGJYAXgC603MADJYAnPMe-ESWS7OrGwDU-B5ItKOudfWuZGa4GPl8hGQAoskdMGC9UASNFgRtRQMCAEABGikMoIwoMsYAwuD4F10kRKunmESiIAcQA) and is ready to be worked on. Thank you for wanting to help.

Rather than do want I suggested in https://github.com/stylelint/stylelint/issues/5997#issuecomment-1080628337, I think it's better if we create a new `isStandardSyntaxKeyframesName` util so that all our `isStandardSyntax*` utilites related to actual types in the spec, [like so](https://drafts.csswg.org/css-animations/#typedef-keyframes-name). 

So:

- create a new `isStandardSyntaxKeyframesName` utility that uses the [hasInterpolation](https://github.com/stylelint/stylelint/blob/main/lib/utils/hasInterpolation.js) utility
- use that `isStandardSyntaxKeyframesName` utility in the `keyframes-name-pattern` rule
